common/thorhelper/thorxmlwrite.cpp

```diff
@@ -70,6 +70,30 @@ void CommonXmlWriter::closeTag()
     flush(false);
 }
 
+void CommonXmlWriter::cutFrom(IInterface *location, StringBuffer& databuf)
+{
+    if (flusher)
+        throwUnexpected();
+
+    CommonXmlPosition *position = dynamic_cast<CommonXmlPosition *>(location);
```

Comment:
Perhaps too big a change now, but feels lke saveLocation / position should be a proper interface with methods to access pos, not dynamic casting up like this.